### ISSUE TITLE
Output Content of the ChatAnthropic Does not match with other providers

### ISSUE THREAD

**Issue by DevDeepakBhattarai** [langchain-ai/langchainjs#6238]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

This is the model selecting code
```typescript
import { env } from "@/env";
import { ChatAnthropic } from "@langchain/anthropic";
import { ChatOpenAI } from "@langchain/openai";
import { ChatGroq } from "@langchain/groq";
import { ChatGoogleGenerativeAI } from "@langchain/google-genai";
import { z } from "zod";
import { ChatVertexAI } from "@langchain/google-vertexai-web";
type Model =
  | ChatOpenAI
  | ChatAnthropic
  | ChatVertexAI
  | ChatGroq
  | ChatGoogleGenerativeAI;
export const AvailableModels = z.enum([
  "gpt",
  "claude",
  "gemini",
  "llama",
  "mistral",
]);
export type AvailableModels = z.infer<typeof AvailableModels>;
export function modelPicker(
  model: z.infer<typeof AvailableModels>,
  stream?: boolean,
) {
  let modelObject: Model;
  switch (model) {
    case "gpt": {
      modelObject = new ChatOpenAI({
        model: "gpt-4o-mini",
        apiKey: env.OPENAI_API_KEY,
        streaming: stream,
        modelKwargs: stream
          ? {
              parallel_tool_calls: false,
            }
          : undefined,
      });

      break;
    }

    case "claude": {
      modelObject = new ChatAnthropic({
        model: "claude-3-haiku-20240307",
        apiKey: env.ANTHROPIC_API_KEY,
        streaming: stream,
      });
      break;
    }

    case "gemini": {
      modelObject = new ChatVertexAI({
        model: "gemini-1.5-flash-001",
        authOptions: {
          credentials: {
            auth_provider_x509_cert_url: env.GOOGLE_AUTH_PROVIDER_X509_CERT_URL,
            auth_uri: env.GOOGLE_AUTH_URI,
            client_email: env.GOOGLE_CLIENT_EMAIL,
            client_id: env.GOOGLE_VERTEX_CLIENT_ID,
            client_x509_cert_url: env.GOOGLE_CLIENT_X509_CERT_URL,
            private_key: env.GOOGLE_PRIVATE_KEY,
            private_key_id: env.GOOGLE_PRIVATE_KEY_ID,
            project_id: env.GOOGLE_PROJECT_ID,
            token_uri: env.GOOGLE_TOKEN_URI,
            type: "service_account",
          },
        },
        temperature: 0,
      });
      break;
    }
    case "llama": {
      modelObject = new ChatGroq({
        apiKey: env.GROQ_API_KEY,
        streaming: stream,
        model: "llama-3.1-70b-versatile",
        temperature: 0.7,
      });
      break;
    }
 }
  return modelObject;
}
```

This is the model invoking code
```typescript

const invokeModel = async (
  state: AgentExecutorState,
  config?: RunnableConfig,
): Promise<Partial<AgentExecutorState>> => {
  const initialPrompt =
    state.model !== "llama" && state.model !== "mistral"
      ? promptWithImage
      : promptWithoutImages;
  const MessageHistoryStore = new UpstashRedisChatMessageHistory({
    sessionId: `${state.userId}-chat-${state.chatId}`, // Or some other unique identifier for the conversation
    client: redis,
  });

  const tools = [
    search_tool,
    weather_tool,
    crypto_tool,
    quiz_tool,
    stock_tool,
    get_tweet_tool,
  ];
  const llm = modelPicker(state.model, true).bindTools(tools);
  const chain = initialPrompt.pipe(llm);
  //  Variable for accumulating the AI response
  let result: AIMessageChunk | undefined = undefined;
  // Variable that is used when the AI gives , text as an output
  let messageStream:
    | ReturnType<typeof createStreamableValue<string>>
    | undefined = undefined;

  // Append loading skeleton on the frontend
  await appendRunnableUI(
    config?.callbacks as CallbackManager,
    <LoadingSkeleton />,
  );

  try {
    // Start the stream
    const stream = await chain.stream(state);

    // Loop through the streamed data
    for await (const chunk of stream) {
      console.log(chunk);
      // When result is already present append the next chunk to it
      if (result) {
        result = result.concat(chunk);
      }
      // When the result is undefined , make it equal to the first chunk
      if (!result) {
        result = chunk;
      }
      //  When the model outputs text, create a streamable value instead of updating the ui.
      // React batches the ui update but , values are not batched
      if (!messageStream && chunk.content) {
        messageStream = createStreamableValue<string>();
        await updateRunnableUI(
          config?.callbacks as CallbackManager,
          <StreamMessage value={messageStream.value}></StreamMessage>,
        );
      }

      // Append the next chunk of the message to already existing messageStream
      if (messageStream && chunk.content) {
        console.log(chunk.content);
        const content = Array.isArray(chunk.content)
          ? chunk.content[0]?.text ?? ""
          : chunk.content;
        messageStream.append(content as string);
      }
    }
    //  Close the messageStream channel once the model output is done
    messageStream?.done();

    console.log("Result", result);
  } catch (e) {
    console.log("There was an error", e);
    return {
      result: undefined,
      toolCall: undefined,
    };
  }

  if (!result) {
    return {
      result: "Sorry there was an error",
    };
  }

  // If the model didn't give any content as an output , remove the loading indicator from the frontend
  if (!result.content) {
    await updateRunnableUI(config?.callbacks as CallbackManager, <></>);
  }

  // Extract tool from the response, because when using gemini , `result.tool_calls` is empty even when the model outputs a tool_call
  const tool_calls =
    result.tool_calls ??
    defaultToolCallParser(result.additional_kwargs.tool_calls ?? [])[0];

  console.log("Tool calls", tool_calls);

  if (tool_calls && tool_calls.length > 0) {
    const tool_call = tool_calls[0]!;
    const toolCall = {
      name: tool_call.name,
      parameters: tool_call.args,
      id: tool_call.id ?? "",
    };
    void MessageHistoryStore.addMessage(
      new AIMessage({
        tool_calls: tool_calls,
        content: tool_calls?.length ? "" : result.content,
      }),
    );
    return {
      toolCall,
      current_run_history: [result],
    };
  }

  void MessageHistoryStore.addMessage(
    new AIMessage({ content: result.content }),
  );

  // const newSummary = await generateSummary(state);
  // void redis.set(`${state.userId}-summary-${state.chatId}`, newSummary);

  return {
    result: result.content as string,
    current_run_history: [result],
    toolCall: undefined,
  };
};
```

### Description

I am trying to build a LLM app with multiple provider support. When ever I use other provider `ChatOpenAI`, `ChatGroq`, `ChatVertexAI` and stream using `chain.stream()`. The `AIMessageChunk` contains `content` that is simply of type `string` but when ever I use `ChatAnthropic` the `content` is an array with the following schema

```json
"content":[
 {
  "type": "text",
  "text": "....."
  },
  {
   "type": "tool_use",
    "id": "toolu_01CG4WGnbowVepRzPFx6d7os",
    "name": "",
    "input": {
        ...agrs
      }
   }
]
```

Is this intentional ? And can this be made to simply include the text rather than whole array. 

If the `tool_use` type is present it should simply appear in `tool_call_chunks`. 

### System Info

windows 10
node 18 
pnpm 9.4

**Comment by bracesproul** [comment]
Hey @DevDeepakBhattarai the `content` field is meant to closely match the content returned from the model provider, but in a LangChain compatible format, e.g plain string, or `type: "text", text: "..."`

The `tool_use` field is LangChain specific and contains any tool calls the model might respond with. I just checked with anthropic and it is including the tool calls inside the `tool_call_chunks` field. Is there another issue?

**Comment by bracesproul** [comment]
I'm going to close this issue now. Let me know if you experience any issues with this!

**Comment by DevDeepakBhattarai** [comment]
Thanks for the explanation. I was using multiple providers , so whenever I stored the result from the AI , it would be stored in the array format causing the webapp to break.  Now I have made changes that check if the content is of type array and handle the result in a different way.
